microprofile/connectors/kafka/src/main/java/io/helidon/microprofile/connectors/kafka/BasicKafkaConsumer.java

```diff
@@ -0,0 +1,207 @@
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.errors.WakeupException;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+/**
+ * Basic Kafka consumer covering basic use-cases.
+ * Configurable by Helidon {@link io.helidon.config.Config Config},
+ * For more info about configuration see {@link KafkaConfigProperties}
+ *
+ * @param <K> Key type
+ * @param <V> Value type
+ * @see KafkaConfigProperties
+ * @see io.helidon.config.Config
+ */
+class BasicKafkaConsumer<K, V> implements Closeable {
+
+    private static final Logger LOGGER = Logger.getLogger(BasicKafkaConsumer.class.getName());
+    private static final String POOL_TIMEOUT = "pool.timeout";
+    private static final String PERIOD_EXECUTIONS = "period.executions";
+    private final KafkaConfigProperties properties;
+    private final Config config;
+    // We need this flag to avoid this task is executed more than one time at the same time by ScheduledExecutorService
+    private final AtomicBoolean running = new AtomicBoolean(false);
+    private final PartitionsAssignedLatch partitionsAssignedLatch = new PartitionsAssignedLatch();
+    private final List<String> topicNameList;
+    // It is not thread safe. It needs to be closed in the same thread it reads events.
+    // We need to keep the reference here to be able to wake up from pooling when shuting down
+    private final KafkaConsumer<K, V> consumer;
+    private final ScheduledExecutorService scheduler;
+
+    /**
+     * Kafka consumer created from {@link io.helidon.config.Config config}
+     * see configuration {@link KafkaConfigProperties example}.
+     *
+     * @param config Helidon {@link io.helidon.config.Config config}
+     * @param scheduler Helidon {@link java.util.concurrent.ScheduledExecutorService scheduler}
+     */
+    BasicKafkaConsumer(Config config, ScheduledExecutorService scheduler) {
+        this.config = config;
+        this.properties = new KafkaConfigProperties(config);
+        this.topicNameList = properties.getTopicNameList();
+        this.consumer = new KafkaConsumer<>(properties);
+        this.scheduler = scheduler;
+    }
+
+    /**
+     * Create publisher builder.
+     *
+     * @return {@link org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder}
+     */
+    public PublisherBuilder<? extends Message<?>> createPushPublisherBuilder() {
+        return ReactiveStreams.fromPublisher(new BasicPublisher<K, V>(subscriber -> {
+            subscriber.onSubscribe(new Subscription() {
+                @Override
+                public void request(long n) {
+                    // Pushing Kafka consumer doesn't support requests.
+                }
```

Comment:
Uf this was my super bad idea, we have to do something about this, it basically ignores backpressure. Something like this would be much better:
https://github.com/oracle/helidon/blob/1e5ae594bc356ecd1283e487a7e7f85e26355ee9/microprofile/reactive-streams/src/main/java/io/helidon/microprofile/reactive/EmittingPublisher.java

But that depends on protected RS with SequentialSubscriber,
I expect David to remove SequentialSubscriber from RS implemetation in #1511 so it gets little more complicated then.

Reply:
I integrated the EmittingSubscriber